ibridges/data_operations.py

```diff
@@ -360,3 +369,38 @@ def create_collection(session: Session,
         raise PermissionError(
                 f"While creating collection at '{coll_path}': iRODS server forbids action."
               ) from exc
+
+
+def perform_operations(session: Session, operations: dict, ignore_err: bool=False):
```

Comment:
Ver handy, this can be used in the GUI to implements sync that emits the current state of the sync!